build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java

```diff
@@ -203,18 +207,34 @@ public void execute() {
                 tags.forEach(tag -> spec.args("--tag", tag));
 
                 parameters.getBuildArgs().get().forEach((k, v) -> spec.args("--build-arg", k + "=" + v));
+
+                if (parameters.getPush().getOrElse(false)) {
```

Comment:
allow pushing directly from here. Makes it easier to push multiple images within a gradle build without the need of doing as part of a script after the build as we use to do for distribution images